src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -47,12 +47,12 @@ protected override bool ReceiveRecover(object message)
             if (message is EntityStarted)
             {
                 var started = (EntityStarted)message;
-                State = new ShardState(State.Entries.Add(started.EntityId));
+                State = new ShardState(State.Entries?.Add(started.EntityId));
```

Comment:
You're absolutely right. Changing `ShardState` to a class resolved all my issues here.

I can't foresee any problems with this change. Can you?